docs/Custom-brimcap-load-Config.md

```diff
@@ -0,0 +1,433 @@
+```
+
+* **Note**: The `zdeps` directory that contains the `zed` and `brimcap`
+binaries varies per platform.
+
+|**OS Platform**|**Location**|
+|---------------|------------|
+| **Windows**   | `%USERPROFILE%\AppData\Local\Programs\Brim\resources\app.asar.unpacked\zdeps` |
+| **macOS**     | `/Applications/Brim.app/Contents/Resources/app.asar.unpacked/zdeps` |
+| **Linux**     | `/opt/Brim/resources/app.asar.unpacked/zdeps` |
+
+If successful, the new pool will appear in Brim, allowing you to browse the
+logs and open flows from the pcap via the **Packets** button.
+
+![NetFlow Pool](media/Custom-Zeek-Suricata-Pool.png)
+
+The same `brimcap load` command line can be used to incrementally add more logs
+to the same pool for additional pcaps, which was not possible pre-Brimcap. The
+setting in the Brim **Preferences** for the **Brimcap Custom YAML Path** can
+also be pointed at the path to this configuration file, which will cause it to
+be invoked when you open or drag pcap files into Brim.
+
+In examining the example Brimcap YAML, we see at the top that we've defined two
+`analyzers`.
+
+```
+analyzers:
+  - cmd: zeek-wrapper.sh
+  - cmd: suricata-wrapper.sh
+```
+
+A Brimcap analyzer has the following characteristics:
+
+1. It expects pcap input to be streamed to it via standard input (stdin)
+2. It's expected to output log files that can be further processed as soon as
+lines are appended to them (i.e. `tail` could process them)
+3. Additional per-analyzer options can be used to affect which generated logs
+are loaded and what additional processing is performed on them
+
+The analyzer invoked by Brimcap is a wrapper script as referenced in the YAML.
+In addition to taking input on stdin, it also sets Zeek to ignore checksums
+(since these are often set incorrectly on pcaps) and we also disable a couple
+of the excess log outputs. This is very similar to the Zeek Runner script that
+was included with Brim `v0.24.0`.
+
+```
+$ cat zeek-wrapper.sh 
+#!/bin/bash
+zeek -C -r - --exec "event zeek_init() { Log::disable_stream(PacketFilter::LOG); Log::disable_stream(LoadedScripts::LOG); }" local
+```
+
+We use a similar wrapper for Suricata, but here we perform some additional
+steps to work around a known Suricata bug [#4106](https://redmine.openinfosecfoundation.org/issues/4106).
+This bug causes EVE JSON events to be output that contain duplicate keys, which
+Zed is not currently equipped to tolerate ([zed/2523](https://github.com/brimdata/zed/issues/2523)). We therefore pipe the EVE JSON data through [`jq`](https://stedolan.github.io/jq/),
+which has the effect of deduplicating these references. Shell pipeline
+additions such that this illustrate the kinds of processing you may want to
+invoke in your Brimcap customizations, as this may prove useful not just for
+working around bugs but also for providing enhanced functionality.
+
+```
+$ cat suricata-wrapper.sh 
+#!/bin/bash
+suricata -r /dev/stdin
+cat eve.json | jq -c . > deduped-eve.json
```

Comment:
This example is hazardous until #71 is fixed. tl;dr - If `jq` is not installed for some reason, Brimcap hangs when using this config.

Reply:
#71 is not fixed, but the `jq` hack is gone, so I'm marking this one as resolved.